src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -283,7 +306,35 @@ void UpdateRootHashesMultiThread()
                     return state;
                 }
                 StorageTree storageTree = kvp.Value;
-                storageTree.UpdateRootHash(canBeParallel: false);
+                ref var dict = ref CollectionsMarshal.GetValueRefOrNullRef(state.changes, kvp.Key);
```

Comment:
This chunk is a direct copy of the single threaded. Could we extract a method maybe?

Reply:
Changed